### PR TITLE
[Merged by Bors] - refactor(Algebra/Module): Move `Module.ofCore` to a MinimalAxioms file, and rename it `ofMinimalAxioms`

### PR DESCRIPTION
**Pull request by martincmartin**
This makes it consistent with Ring, Field and Group.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by martincmartin** [comment]
Should I rename the structure `Module.Core` to `Module.MinimalAxioms`?  Or just get rid of it, to be consistent with the other `ofMinimalAxioms` functions?

File: Mathlib/Algebra/Module/MinimalAxioms.lean [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/-
+Copyright (c) 2015 Nathaniel Thomas. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nathaniel Thomas, Jeremy Avigad, Johannes Hölzl, Mario Carneiro, Martin C. Martin
+-/
+import Mathlib.Algebra.Module.Basic
+
+/-!
+# Minimal Axioms for a Module.
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/MinimalAxioms.lean:9]:
```suggestion
# Minimal Axioms for a Module
```

File: Mathlib/Algebra/Module/MinimalAxioms.lean [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+This file defines a constructor to define a `Module` structure on a Type with an
+AddCommGroup, while proving a minimum number of equalities.
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/MinimalAxioms.lean:12]:
```suggestion
`AddCommGroup`, while proving a minimum number of equalities.
```

**Review by eric-wieser — APPROVED** [review]
LGTM, thanks!

Can you update the PR title to reflect the [commit conventions](https://leanprover-community.github.io/contribute/commit.html)?

**Comment by eric-wieser** [comment]
> Or just get rid of it, to be consistent with the other `ofMinimalAxioms` functions?

This sounds like a good idea to me. The original design is vestigial from the refactor in https://github.com/leanprover-community/mathlib/pull/456, so I don't think we need to be attached to it.

**Comment by martincmartin** [comment]
@eric-wieser I believe I've addressed all your comments.  Please let me know if you'd like me to change anything else before merging.  And thanks for reviewing and your helpful comments!

File: Mathlib/Algebra/Module/MinimalAxioms.lean [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+* `Module.ofMinimalAxioms`: Define a `Module` structure on a Type with an
+  AddCommGroup by proving a minimized set of axioms
+
+-/
+
+universe u v
+
+/-- Define a `Module` structure on a Type by proving a minimized set of axioms. -/
+def Module.ofMinimalAxioms {R : Type u} {M : Type v} [Semiring R] [AddCommGroup M] [SMul R M]
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Algebra/Module/MinimalAxioms.lean:24]: Using `reducible` instead for consistency is also fine.

[the remark this answers is not in the document]
Comment (eric-wieser) [reply Mathlib/Algebra/Module/MinimalAxioms.lean:24]:
```suggestion
@[reducible]
def Module.ofMinimalAxioms {R : Type u} {M : Type v} [Semiring R] [AddCommGroup M] [SMul R M]
```

**Comment by eric-wieser** [comment]
bors merge

Thanks!

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**